src/openproduct/utils/oidc_drf_middleware.py

```diff
@@ -17,14 +20,24 @@ def authenticate(self, request):
 
         try:
             return super().authenticate(request)
+        except AssertionError as exc:
+            logger.exception(
+                "OIDC authentication failed due to assertion error: %s", exc
+            )
+            raise AuthenticationFailed(
+                "OIDC authentication failed: authentication is not properly configured."
+            )
         except HTTPError as exc:
             resp = exc.response
             msg = "OIDC authentication failed with status code: {}".format(
                 resp.status_code
             )
-
             if "www-authenticate" in resp.headers:
                 data = parse_www_authenticate_header(resp.headers["www-authenticate"])
-                msg = f"{msg} www_authenticate: {data}"
+                error_description = data.get(
+                    "error_description", "no error description in www-authenticate"
+                )
+                msg = "{} www_authenticate: {}".format(msg, error_description)
 
+            logger.exception(msg)
```

Comment:
Let's make sure it's an event and could you also add it to `logging.rst`? The exc_info should be automatically included because of the `.exception` call
```suggestion
            logger.exception("oidc_authentication_failed")
```